src/panels/config/automation/ha-automation-picker.ts

```diff
@@ -898,6 +906,12 @@ class HaAutomationPicker extends SubscribeMixin(LitElement) {
     `;
   }
 
+  private _handleMenuClose(ev) {
+    if (ev.detail.reason.key === "Escape") return;
+    const action = ev.currentTarget.action.bind(this);
```

Comment:
I wonder if we should use a switch/case with a string action to call the right function instead of binding the function on the element.

Reply:
Instead of using `target` or `currentTarget`, use `initiator` defined in the custom event detail.  Then I wonder if the key/reason check is even necessary as hopefully that would be `null` or `undefined` unless the user actually took the action with space or enter?

Reply:
The initiator is still `ha-menu-item` when canceling the menu with Escape key.